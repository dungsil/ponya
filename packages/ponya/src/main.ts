import { createApp } from 'vue-demi'
import App from './app.vue'

import 'uno.css'

createApp(App)
  .mount('#top')
