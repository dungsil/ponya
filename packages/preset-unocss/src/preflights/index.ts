import type { Preflight } from '@unocss/core'
import { Theme } from '@unocss/preset-mini'

// noinspection CssUnknownProperty
export const preflights: Preflight<Theme>[] = [
  {
    layer: 'preflights',
    // language=CSS
    getCSS: ({ theme }) => `
        :root {
            
        }
    `
  }
]
