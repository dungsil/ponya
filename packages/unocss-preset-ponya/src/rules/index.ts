import type { Rule } from '@unocss/core'
import type { Theme } from '../theme'
import { wh, minOrMaxWh, minMaxWh, mb, mbXy, display, flexDirections } from './layouts'


export const rules: Rule<Theme>[] = [
  // Layouts
  display,
  flexDirections,

  wh, minOrMaxWh, minMaxWh,
  mb, mbXy,

  // Typography
]
