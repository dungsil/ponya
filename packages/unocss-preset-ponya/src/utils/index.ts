export * from './css'
export * from './colors'
