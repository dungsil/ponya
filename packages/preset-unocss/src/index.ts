import { type Preset, Rule } from 'unocss'
import { type Theme } from '@unocss/preset-mini'
import { rules } from '@unocss/preset-mini/rules'

import { theme } from './theme'

/**
 * Ponya preset
 */
export function presetPonya(): Preset<Theme> {


  return {
    name: '@ponya-ui/preset-unocss',
    theme,
    rules: [
      // base rules
      ...rules as Rule<Theme>[]
    ],
  }
}
