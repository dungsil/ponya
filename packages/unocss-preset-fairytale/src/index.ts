import { type Preset } from 'unocss'

import { theme, type Theme } from './theme'
import { preflights } from './preflights'
import { rules } from './rules'

/**
 * Fairytale preset
 */
export function presetFairytale(): Preset<Theme> {

  return {
    name: 'unocss-preset-fairytale',
    theme,
    preflights,
    rules,
  }
}
