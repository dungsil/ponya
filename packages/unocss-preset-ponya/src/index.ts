import { type Preset } from 'unocss'

import { theme, type Theme } from './theme'
import { preflights } from './preflights'
import { rules } from './rules'

/**
 * Ponya preset
 */
export function presetPonya(): Preset<Theme> {

  return {
    name: 'unocss-preset-ponya',
    theme,
    preflights,
    rules,
  }
}
