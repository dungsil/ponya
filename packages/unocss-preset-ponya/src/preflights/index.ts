import type { Preflight } from '@unocss/core'
import type { Theme } from '../theme'
import { colorResolver } from '../utils'

// noinspection CssUnknownProperty
export const preflights: Preflight<Theme>[] = [
  {
    layer: 'preflights',
    // language=CSS
    getCSS: ({ theme }) =>  `
html[data-color-scheme="auto"] {
    color-scheme: normal;    
}
html[data-color-scheme="light"] {
    color-scheme: light;
}
html[data-color-scheme="dark"] {
    color-scheme: dark;
}

body {
  font-family: ${theme.font.family[theme.base.font.family]};
  font-weight: ${theme.font.weight[theme.base.font.weight]};
  font-size: ${theme.font.size[theme.base.font.size]};
  line-height: ${theme.lineHeight[theme.base.lineHeight]};
  letter-spacing: ${theme.letterSpacing[theme.base.letterSpacing]};
  color: ${colorResolver(theme, theme.base.font.color)};
}
    `
  }
]
