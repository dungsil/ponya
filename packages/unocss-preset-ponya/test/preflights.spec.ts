import { describe, test, expect } from 'vitest'
import { createGenerator } from 'unocss'
import { presetPonya } from '../src'

const uno = createGenerator({
  // @ts-ignore
  presets: [presetPonya()]
})

describe('preflights', () => {
  test('generate test', async () => {
    const {css} = await uno.generate('', { preflights: true })
    expect(css).toMatchSnapshot()
  })
})
