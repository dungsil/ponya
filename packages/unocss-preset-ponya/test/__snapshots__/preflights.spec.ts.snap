// Vitest Snapshot v1

exports[`preflights > generate test 1`] = `
"/* layer: preflights */

body {
  font-family: Pretendard, -apple-system, BlinkMacSystemFont, system-ui, Roboto, 'Helvetica Neue', 'Segoe UI', 'Apple SD Gothic Neo', 'Noto Sans KR', 'Malgun Gothic', 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.0625em;
  color: #343a40;
}
    "
`;
